oletools/util/log_helper/log_helper.py

```diff
@@ -0,0 +1,208 @@
+# 2018-02-07 v0.04 SA: - fixed control of handlers propagation
+# 2018-04-23 v0.05 SA: - refactored the whole logger to use an OOP approach
+
+# -----------------------------------------------------------------------------
+# TODO:
+
+
+from __future__ import print_function
+from ._json_formatter import JsonFormatter
+from ._null_handler import NullHandler
+from ._logger_class import OletoolsLogger
+from . import _root_logger_wrapper
+import logging
+import sys
+
+
+LOG_LEVELS = {
+    'debug': logging.DEBUG,
+    'info': logging.INFO,
+    'warning': logging.WARNING,
+    'error': logging.ERROR,
+    'critical': logging.CRITICAL
+}
+
+DEFAULT_LOGGER_NAME = 'oletools'
+DEFAULT_MESSAGE_FORMAT = '%(levelname)-8s %(message)s'
+
+
+class LogHelper:
+    def __init__(self):
+        self._all_names = set()  # set so we do not have duplicates
+        self._use_json = False
+        self._is_enabled = False
+        self._json_formatter = JsonFormatter()
+        logging.setLoggerClass(OletoolsLogger)
+
+    def get_or_create_logger(self, name=DEFAULT_LOGGER_NAME, level=logging.WARNING):
+        """
+        Get a logger or create one if it doesn't exist.
+        To keep the same behavior as `logging.getLogger`, we add a StreamHandler
+        that logs by default to stderr
+        """
+        return self._get_or_create_logger(name, level, logging.StreamHandler())
+
+    def get_or_create_silent_logger(self, name=DEFAULT_LOGGER_NAME, level=logging.CRITICAL + 1):
+        """
+        Get a logger or create one if it doesn't exist, setting a NullHandler
+        (to avoid printing to the console) as the handler if that's the case.
+        By default we also use a higher logging level so every message will
+        be ignored.
+        This is useful when we don't want to print anything when the logger
+        is not configured by the main application.
+
+        Python 2.7 has logging.NullHandler, but this is necessary for 2.6:
+        git pus https://docs.python.org/2.6/library/logging.html#configuring-logging-for-a-library
+        """
+        if sys.version_info >= (2, 6):
+            handler = logging.NullHandler
+        else:
+            handler = NullHandler
+
+        return self._get_or_create_logger(name, level, handler())
+
+    def enable_logging(self, use_json, level, log_format=DEFAULT_MESSAGE_FORMAT, stream=None):
+        """ called from main after parsing arguments """
+        if self._is_enabled:
+            raise ValueError('re-enabling logging. Not sure whether that is ok...')
+
+        log_level = LOG_LEVELS[level]
+        logging.basicConfig(level=log_level, format=log_format, stream=stream)
+        self._is_enabled = True
+
+        self._use_json = use_json
+
+        sys.excepthook = self._log_except_hook
+
+        # since there could be loggers already created we go through all of them
+        # and set their formatters to our custom Json formatter
```

Comment:
what if oletools is imported into an application that has specific logging needs? Wouldn't that be too intrusive? 
As a library, I think oletools should not modify the runtime environment (for example other loggers) too much. This caused some bugs in the past.

Reply:
Indeed I think it would be better only when we control the main app. Or maybe there is a way to check which loggers need that formatter or not? (to avoid touching non-oletools loggers)

Reply:
This is a good point. I thought that instead of replacing the whole exception handling function we could just wrap it with another function that prints the closing bracket, but as Christian pointed out in our private discussion this is still a bit intrusive since `sys.excepthook` is being modified anyway.

@decalage2 as for the second option you've given, currently either every logger created by the log module is set to JSON or none is. We also keep track of which loggers we created to avoid messing up other loggers.

Maybe we could add a decorator to the main() functions that would handle this check (e.g. if calling as script, just overwrite the generic exception handler) or set this inside the `if __name__ == '__main__'` check.

Reply:
Correct me if I am wrong:
(1) we assume that we are being run "as script" (as opposed to being imported as library from other application) if (and only if) the main function runs
(2) Only in the main function we call LogHelper.enable_logging(...)
(3) Only in LogHelper.enable_logging(), the sys.excepthook is modified.
--> If (and only if) we are running as script, the sys.excepthook is modified.
That is what we wanted, is it not?